### DISCUSSION TITLE
Help wanted. Github OAuth works fine locally but stops working after deployment on Vercel

### DISCUSSION THREAD

**Discussion by RushilJalal** [vercel/next.js#67278] [Help] [answered]
### Summary

Logs on vercel

![image](https://github.com/nextauthjs/next-auth/assets/54212357/1efa56aa-7351-4a35-b99a-b047e3e35bc7)

The website 

![image](https://github.com/nextauthjs/next-auth/assets/54212357/5977f8bf-83c5-45c2-9fb0-a60c073e4a2b)

On clicking the sign in button

![image](https://github.com/nextauthjs/next-auth/assets/54212357/2c960274-2dbb-48ed-9969-307045b56aa6)

reloading the page. it looks like everything is working fine, until after logging i get this error

![image](https://github.com/nextauthjs/next-auth/assets/54212357/c264acba-52a2-4a8b-98e7-e68ee28944f5)

Github OAuth app

![image](https://github.com/nextauthjs/next-auth/assets/54212357/8baceb58-fc0f-4eeb-ba8d-240f5b668946)

Env vars on Vercel

![image](https://github.com/nextauthjs/next-auth/assets/54212357/1af26f45-6be7-4588-853e-c3908973614d)

Here's my repo : https://github.com/RushilJalal/rushil-ecommerce-store

Please feel free to ask me any questions. I'm a beginner and I would love to make this work.

### Additional information

```js
All error messages are in the above section

On running `next info`, I get an error
```

### Example

https://github.com/RushilJalal/rushil-ecommerce-store

**Comment by levis-creator** [comment]
For me i solved the problem by deploying the project in small chunks and it started working

**Reply by RushilJalal** [reply]
how do i do this? Also wdym?

**Comment by RushilJalal** [comment]
Update : 

page after clicking sign in

![image](https://github.com/vercel/next.js/assets/54212357/5139f1f7-51ce-4ce1-bbca-ad365676139f)

I refresh the page...AND IT WORKS

![image](https://github.com/vercel/next.js/assets/54212357/ebac0445-d956-4d9e-ad58-e0395a24a804)

I have no idea why I am getting this 404 error
I would love some help

**Reply by levis-creator** [reply]
Are you redirecting your page when signin is successful? using github?

**Comment by RushilJalal** [comment, marked as answer]
The error has been found. It was the NextJS version after all.
Read here: https://github.com/nextauthjs/next-auth/issues/10585
Reverting back to `v14.1.2` fixed it.
